### ISSUE TITLE
Scope: Use Rust apps?

### ISSUE THREAD

**Issue by idnovic** [ublue-os/cosmic#74]
Cosmic is a very new desktop and very bare bone.

But it is programmed in Rust.
Should the scope of the project recommend native rust apps to be included?

Example this flatpak as foto manager
https://flathub.org/apps/app.fotema.Fotema

Suggestion:
Change scope to reflect the preference of rust apps over Gtk or QT apps
if a similar functionality can be achieved with a rust counterpart.

Cosmic native > rust app > Gtk app or Qt app.

If this change is accepted I can take the time to find recommendable rust apps native and flatpak.

**Comment by ryanabx** [comment]
I think this would be part of a larger question about this image. Do we want this image to only provide the COSMIC desktop, or do we want this image to be a more opinionated experience akin to Bluefin and Aurora?

I personally think that the base cosmic image we provide is important for further downstreams to use, since Fedora doesn't have a COSMIC base image; however, I also think it'd be nice for us to in addition provide an opinionated image with a full experience, which could definitely include rust applications.

**Comment by p5** [comment]
> I personally think that the base cosmic image we provide is important for further downstreams to use, since Fedora doesn't have a COSMIC base image

I completely agree.  We should be treating this image as if it were provided by Fedora, then the opinionated changes will be added downstream (for example in Bazzifin) or external custom images.

**Comment by ahoneybun** [comment]
My current plan is to only install COSMIC community apps that have flatpaks with [Supernova](https://github.com/ahoneybun/supernova) but I'm still working on figuring out the build system and such.

**Comment by idnovic** [comment]
I have a few open thoughts on this and would like to share them. 

I think a middle ground between bare bone and opinionated. 

Basically offer OOTB experience for the most important tasks. 
Photo
Video
Music
Web
File Management

With a little bit of opinionated changes like
blueman for bluetooth
network manager gui
Pipewire gui
Archiv manager

This way the image stays bare bone but it also offers needed base functionality until cosmic integrates these features. 

Try to send a file via bluetooth. 
Try to change the bluetooth profile of headphones. 
Try to change the firewalld zone for an interface. 
Try to set a default file association for xyz. 
...

their is barebone and their is barely useable. 
Cosmic as alpha is the later. And that is okay and to be expected. 

Is every downstream project supposed to layer the same base packages for base functionality? 

What should be added is the gnome keyring. 
At least until cosmic offers an integration or an own implementation. 
Because without it apps start to save login details in a non favourable manner. (gnome keyring is not auto unlocked atm, even if pkgs have been added).

**Comment by ahoneybun** [comment]
The network parts should land soonish though I also don't know about including somethings since COSMIC hasn't had it's first release yet.

**Comment by idnovic** [comment]
> My current plan is to only install COSMIC community apps that have flatpaks with [Supernova](https://github.com/ahoneybun/supernova) but I'm still working on figuring out the build system and such.

Just in case you want to check out bluebuild for your image. 
You can take a look at my repo. 
https://github.com/idnovic/secureblue-extended

https://blue-build.org

My image is very customised but I use nearly every feature of the build system. 

Took me a few days to sort everything out. 
I can recommend blue build for personal images over upstream buildha + podman. 

That said I decided to keep gnome for the moment and follow cosmic desktop development.

**Comment by ahoneybun** [comment]
My image is forked from the cosmic image but once it is all in Fedora 41 I plan to use the template from ublue instead I think. It seems to be updated and better supported for features.

**Comment by bhh32** [comment]
I agree that this image should be a vanilla Fedora COSMIC experience. Having it opinionated on top of a brand new DE may be too much. Personally, I'm using the Pop!OS alpha and vanilla COSMIC alpha on top of Fedora 40 right now and it's pretty great as is. It's going to be even better when it's actually done.
